src/hooks/useLocalStorage.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+
+const useLocalStorage = (key, defaultValue) => {
+  const [localStorageValue, setLocalStorageValue] = useState(() => {
+    try {
+      const value = localStorage.getItem(key);
+      if (value) {
+        return JSON.parse(value);
+      } else {
+        localStorage.setItem(key, JSON.stringify(defaultValue));
+        return defaultValue;
+      }
+    } catch (error) {
+      localStorage.setItem(key, JSON.stringify(defaultValue));
+      return defaultValue;
+    }
+  });
+
+  const setLocalStorageStateValue = (valueOrFn) => {
+    let newValue;
+    if (typeof valueOrFn === "function") {
```

Comment:
@NemishGorasiya 
I don't think this is good way to pass value in function. so remove it. make it simple